src/tools/android/java/com/google/devtools/build/android/desugar/dependencies/DesugarDependencyInfoWrapper.java

```diff
@@ -0,0 +1,67 @@
+package com.google.devtools.build.android.desugar.dependencies;
+
+import com.google.devtools.build.android.desugar.proto.DesugarDeps;
+
+import java.util.ArrayList;
+import java.util.Comparator;
+
+/**
+ * a wrapper that is responsible to sort desugar dependency collector items since these items will
+ * be add to the jar file as metadata it should be deterministic so that it won't cause cache misses
+ */
+public class DesugarDependencyInfoWrapper {
```

Comment:
I think we could simplify things by merging this into the MetadataCollector class itself -- i.e. MetadataCollector can have the the lists, and add to those, then have a private build() method that sorts the lists, and creates the DesugarDepsInfo proto

Reply:
make sense